embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryServiceTest.kt

```diff
@@ -4,156 +4,164 @@ import com.google.common.util.concurrent.MoreExecutors
-
-        val observed = deliveryCacheManager.saveSessionRequests.last()
-        assertEquals(Pair(sessionMessage, JVM_CRASH), observed)
+    fun `send session synchronously on crash successfully`() {
+        deliveryService.sendSession(sessionMessage, SessionSnapshotType.JVM_CRASH)
+        assertTrue(apiService.sessionRequests.contains(sessionMessage))
+        assertEquals(1, apiService.futureGetCount)
+        assertNull(cacheService.loadObject(sessionFileName, SessionMessage::class.java))
     }
 
     @Test
     fun `if no previous cached session then send previous cached sessions should not send anything`() {
-        initializeDeliveryService()
         deliveryService.sendCachedSessions(null, sessionIdTracker)
         assertTrue(apiService.sessionRequests.isEmpty())
     }
 
     @Test
     fun `send previously cached sessions successfully`() {
-        initializeDeliveryService()
-        deliveryCacheManager.addCachedSessions(sessionMessage, anotherMessage)
-
+        assertNotNull(cacheService.writeSession(sessionFileName, sessionMessage))
+        assertNotNull(cacheService.writeSession(anotherMessageFileName, anotherMessage))
         deliveryService.sendCachedSessions(null, sessionIdTracker)
-        assertEquals(listOf(sessionMessage, anotherMessage), apiService.sessionRequests)
+        assertTrue(apiService.sessionRequests.contains(sessionMessage))
+        assertTrue(apiService.sessionRequests.contains(anotherMessage))
+        assertEquals(2, apiService.sessionRequests.size)
+        val sessionMap = apiService.sessionRequests.associateBy { it.session.sessionId }
+        assertEquals(sessionMessage, sessionMap[sessionMessage.session.sessionId])
+        assertEquals(anotherMessage, sessionMap[anotherMessage.session.sessionId])
     }
 
     @Test
     fun `ignore current session when sending previously cached sessions`() {
-        initializeDeliveryService()
-        deliveryCacheManager.addCachedSessions(sessionMessage, anotherMessage)
+        assertNotNull(cacheService.writeSession(sessionFileName, sessionMessage))
+        assertNotNull(cacheService.writeSession(anotherMessageFileName, anotherMessage))
         sessionIdTracker.sessionId = anotherMessage.session.sessionId
         deliveryService.sendCachedSessions(null, sessionIdTracker)
         assertEquals(listOf(sessionMessage), apiService.sessionRequests)
     }
 
     @Test
     fun `if an exception is thrown while sending cached session then sendCachedSession should not crash`() {
-        initializeDeliveryService()
-        deliveryCacheManager.addCachedSessions(sessionMessage)
+        assertNotNull(cacheService.writeSession(sessionFileName, sessionMessage))
         apiService.throwExceptionSendSession = true
         deliveryService.sendCachedSessions(null, sessionIdTracker)
         assertTrue(apiService.sessionRequests.isEmpty())
     }
 
-    @Test
-    fun `send session end with crash`() {
```

Comment:
This wasn't doing what it was purporting and is basically duplicating `send session synchronously on crash successfully`() now.